### PR TITLE
[sc-9421] New sync page 

### PR DESCRIPTION
**Pull request by mpellerin42**

### CODE REVIEW

File: libs/sync/src/assets/i18n/en.json [hunk 1]
```diff
@@ -1,109 +1,138 @@
...
+	"sync.add-page.description.folder": "Upload and synchronize content from your desktop folders.",
+	"sync.add-page.description.gdrive": "Upload and synchronize content from selected Google Drive folders.",
+	"sync.add-page.description.onedrive": "Upload and synchronize content from selected Onedrive folders.",
+	"sync.add-page.description.rss": "Upload and synchronize content from a RSS source.",
+	"sync.add-page.description.sharepoint": "Upload and synchronize content from selected Sharepoint folders.",
+	"sync.add-page.description.sitemap": "Upload and synchronize web pages content from a sitemap.",
+	"sync.add-page.filter-resources.description": "We will only sync the resources matching the selection.",
+	"sync.add-page.filter-resources.file-extension.exclude": "Exclude",
+	"sync.add-page.filter-resources.file-extension.include": "Include",
+	"sync.add-page.filter-resources.file-extension.info": "Separate the file extensions you want to include or exclude with commas.",
```
Comment (ebrehault) [inline libs/sync/src/assets/i18n/en.json:24]: easier to read
```suggestion
	"sync.add-page.filter-resources.file-extension.info": "Separate with commas the file extensions you want to include or exclude.",
```

File: libs/sync/src/lib/sync-details-page/sync-settings/sync-settings.component.html [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+      [itemTitle]="section.title"
+      [description]="section.description || ''">
+      <div class="sync-settings-container">
+        @for (field of section.fields; track field.id) {
+          <div class="body-s settings-item">
+            <label>{{ field.label | translate }}</label>
+            @if (field.type !== 'table') {
+              <div>{{ sync.connector.parameters[field.id] || '_' }}</div>
+            } @else if (sync.connector.parameters[field.id].lenght > 0) {
+              <app-parameters-table
```
[the remark this answers is not in the document]
Comment (mpellerin42) [reply libs/sync/src/lib/sync-details-page/sync-settings/sync-settings.component.html:94]: This was working before, I probably broke it in my last changes. Let me check and fix this. Thanks for catching it up

File: libs/common/src/lib/upload/upload-data/upload-layout.component.ts
Comment (operramon) [inline libs/common/src/lib/upload/upload-data/upload-layout.component.ts:1]: could this component be deleted ?

Comment (mpellerin42) [reply libs/common/src/lib/upload/upload-data/upload-layout.component.ts:1]: yes probably, you're right

File: libs/sync/src/assets/i18n/en.json [hunk 3]
```diff
@@ -1,109 +1,138 @@
...
+	"sync.home-page.add-sources.title": "Add sources",
+	"sync.home-page.description": "Upload and synchronize content from third-party sources.",
+	"sync.home-page.setup-server.description": "The Nuclia Sync Agent is an application that allows you to synchronize a local folder or a third-party data source. Its purpose is to manage all the 3rd-party storage services authentication and data transfer on your premises for privacy and security reasons.",
+	"sync.home-page.setup-server.on-desktop.download-dropdown": "Download",
+	"sync.home-page.setup-server.on-desktop.radio-help": "To install the sync server locally, <a href=\"https://github.com/nuclia/sync-agent/releases\" target=\"_blank\">download</a> the latest version for your system.",
+	"sync.home-page.setup-server.on-desktop.radio-label": "Use the Nuclia Sync Agent on your desktop",
+	"sync.home-page.setup-server.on-server.radio-help": "To deploy it manually on a server, follow these <a href=\"https://github.com/nuclia/sync-agent/blob/main/server/README.md\" target=\"_blank\">instructions</a>.",
+	"sync.home-page.setup-server.on-server.radio-label": "Deploy it manually on a server",
```
Comment (ebrehault) [inline libs/sync/src/assets/i18n/en.json:122]: I think "Deploy" is not accurate (the user will deploy it once, but then they will use the server, not deploy it every time).
```suggestion
	"sync.home-page.setup-server.on-server.radio-label": "Use a remote server",
```

File: libs/sync/src/assets/i18n/en.json [hunk 4]
```diff
@@ -1,109 +1,138 @@
...
+	"sync.home-page.setup-server.title": "Set up a sync server",
+	"sync.home-page.sync-list.description": "You can manage here you different sources, your sync agent must be active.",
```
Comment (ebrehault) [inline libs/sync/src/assets/i18n/en.json:124]: typo
```suggestion
	"sync.home-page.sync-list.description": "You can manage here your different sources, your sync agent must be active.",
```

File: libs/sync/src/lib/add-sync-page/add-sync-page.component.ts [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+                }),
+                description: this.translate.instant('sync.add-page.confirm-authenticate.description'),
+                confirmLabel: this.translate.instant('sync.add-page.confirm-authenticate.confirm-button'),
+              }).onClose,
+          ),
+          filter((confirmed) => !!confirmed),
+          switchMap(() => this._createSync(syncEntity)),
+        )
+        .subscribe({
+          next: (connector) => this._onSuccessfulCreation(connector, syncEntity),
```
Comment (ebrehault) [inline libs/sync/src/lib/add-sync-page/add-sync-page.component.ts:153]: subscribe in subscribe (`_onSuccessfulCreation` calls `_syncCreationDone` which makes a subscribe)

File: libs/sync/src/lib/add-sync-page/add-sync-page.component.ts [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+
+  save() {
+    if (!this.configuration) {
+      return;
+    }
+    this.saving = true;
+    const syncEntity = this.configuration;
+    if (!this.syncId) {
+      this._createSync(syncEntity).subscribe({
+        next: (connector) => this._onSuccessfulCreation(connector, syncEntity),
```
Comment (ebrehault) [inline libs/sync/src/lib/add-sync-page/add-sync-page.component.ts:194]: subscribe in subscribe

File: libs/sync/src/lib/add-sync-page/add-sync-page.component.ts [hunk 7]
```diff
@@ -0,0 +1,275 @@
...
+        error: (error) => this._errorHandler(error),
+      });
+    } else if (this.folderSelection.length > 0) {
+      const syncId = this.syncId;
+      this.syncService.updateSync(syncId, { foldersToSync: this.folderSelection }, true).subscribe({
+        next: () => {
+          this._syncCreationDone(syncId);
```
Comment (ebrehault) [inline libs/sync/src/lib/add-sync-page/add-sync-page.component.ts:201]: subscribe in subscribe

**Review by ebrehault — APPROVED** [review]

File: libs/sync/src/lib/add-sync-page/add-sync-page.component.ts (hunk 5 above)
Comment (mpellerin42) [reply libs/sync/src/lib/add-sync-page/add-sync-page.component.ts:153]: right, I added `triggerSync` at the end and didn't think about that 🤦 
Fixing

**Review by ebrehault — APPROVED** [review]

**Pull request merged**